localstack/services/apigateway/apigateway_listener.py

```diff
@@ -425,6 +427,34 @@ def invoke_rest_api_integration(api_id, stage, integration, method, path, invoca
             )
             response.headers['content-type'] = APPLICATION_JSON
             return response
+        elif 's3:path/' in uri and method == 'GET':
+            s3 = aws_stack.connect_to_service('s3')
+            uri_match = re.match(S3_REGEX_URI, uri)
+            if uri_match:
+                bucket, object_key = uri_match.group('bucket', 'object')
+                LOGGER.debug('Getting request for bucket %s object %s', bucket, object_key)
+                try:
+                    object = s3.get_object(Bucket=bucket, Key=object_key)
+                except s3.exceptions.NoSuchKey:
+                    msg = 'Object %s not found' % object_key
+                    LOGGER.debug(msg)
+                    return make_error_response(msg, 404)
+
+                headers = aws_stack.mock_aws_request_headers(service='s3')
+
+                if object.get('ContentType'):
+                    headers['Content-Type'] = object['ContentType']
+
+                # stream used so large files do not fill memory
+                response = request_response_stream(
+                    stream=object['Body'],
+                    headers=headers
+                )
+                return response
+            else:
+                msg = 'Request URI does not match s3 specifications'
+                LOGGER.warning(msg)
+                return make_error_response(msg, 500)
```

Comment:
Let's better return a `400` response here, as this is a client error (invalid request).

Reply:
My train of thought here was, that the request that triggers the error may not be invalid, since the uri is specified before the integration, but you are right, its not localstacks fault.

Reply:
Hm, I see your point. Well, I guess in this case 500 would have been legitimate as well - ultimately we could raise the 400 error on _creation_ of the API Gateway integration (if an invalid URL is passed in the configuration). Don't think this is high priority, though - what we have is good for now 👍 . Thanks